File: C/7kyu/Battle of the characters.c
/*
* https://www.codewars.com/kata/595519279be6c575b5000016/train/c
*/

#include <stdio.h>

const char *battle(const char *group_1, const char *group_2);

int main(){

    battle("AAA", "Z");

    return 0;
}


const char *battle(const char *group_1, const char *group_2){

    int sum1 = 0;
    int sum2 = 0;

    for (int i = 0; i < strlen(group_1); i++) sum1 += (int) group_1[i] - 64;
    for (int i = 0; i < strlen(group_2); i++) sum2 += (int) group_2[i] - 64;

    if (sum1 > sum2) return group_1;
    else if (sum2 > sum1) return group_2;
    else return "Tie!";
}




/*
const char *battle (const char *a, const char *b)
{
  int n = 0;
  const char *ra = a, *rb = b;

  while (*a) n += *a++ - 'A' + 1;
  while (*b) n -= *b++ - 'A' + 1;

  return  n >  0 ? ra : n < 0 ? rb : "Tie!";
}
*/


/*
const char *battle (const char *g1, const char *g2) {
  int pon1 = 0, pon2 = 0;

  for(int i = 0; g1[i] != '\0'; i++) pon1 += (g1[i] - 'A') + 1;
  for(int i = 0; g2[i] != '\0'; i++) pon2 += (g2[i] - 'A') + 1;

  return (pon1 > pon2) ? g1 : (pon1 < pon2) ? g2 : "Tie!";
}
*/

File: Python/7kyu/Battle of the characters.py
# https://www.codewars.com/kata/595519279be6c575b5000016/train/python

def battle(x, y):return x if sum([ord(c)-64 for c in x])>sum([ord(c)-64 for c in y]) else y if sum([ord(c)-64 for c in y])>sum([ord(c)-64 for c in x]) else 'Tie!'


if __name__ == "__main__":
        print(battle("AAA", "Z"), "Z", "Unfair fight!")
        print(battle("ONE", "TWO"), "TWO", "Unfair fight!")
        print(battle("ONE", "NEO"), "Tie!", "Unfair fight!")
        print(battle("FOUR", "FIVE"), "FOUR", "Unfair fight!")


"""
battle = lambda x,y,fn = lambda z: sum(ord(c) - 64 for c in z):\
["Tie!",x,y][(fn(x) > fn(y)) - (fn(x) < fn(y))]
"""

"""
def battle(x, y):
    X, Y = sum(ord(i)-64 for i in x), sum(ord(i)-64 for i in y)
    return X > Y and x or X < Y and y or 'Tie!'
"""

"""
battle=lambda x,y,f=lambda s:sum(ord(c)-64for c in s):["Tie!",x,y][(f(x)>f(y))-(f(x)<f(y))]
"""